### PR TITLE
Assert Orange indexes analysis bundle

### PR DESCRIPTION
**Pull request by alex-hancock**
Last commit to be removed after testing.

### CODE REVIEW

**Comment by alex-hancock** [comment]
This test is intended to fail because Orange fails to index the analysis bundle.

File: tests/dataset_runner.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def _assert_data_browser_bundles(self, project_shortname):
         try:
             response_json = self.azul_agent.get_specimen_by_project(project_shortname)
             orange_uuids = {bundle["bundleUuid"] for hit in response_json["hits"] for bundle in hit["bundles"]}
-            assert orange_uuids == set(self.primary_bundle_uuids)
+            assert orange_uuids == set().union(self.primary_bundle_uuids, self.secondary_bundle_uuids)
```
Comment (hannes-ucsc) [inline tests/dataset_runner.py]: I would have preferred either of these:

`{*self.primary_bundle_uuids, *self.secondary_bundle_uuids}`
`set(self.primary_bundle_uuids) | set(self.primary_bundle_uuids)`
`set(self.primary_bundle_uuids).union(self.primary_bundle_uuids)`

File: tests/azul_agent.py [hunk 2]
```diff
@@ -12,6 +12,11 @@ def __init__(self, deployment):
     def get_specimen_by_project(self, project_shortname):
         filters = {"file": {"project": {"is": [project_shortname]}}}
         path = '/repository/specimens?filters=' + urllib.parse.quote(json.dumps(filters))
-        response = requests.get(self.azul_url + path)
+        azul_session = requests.Session()
```
Comment (hannes-ucsc) [inline tests/azul_agent.py]: Ideally, I'd want to stash the session in an instance attribute.

File: tests/azul_agent.py [hunk 3]
```diff
@@ -12,6 +12,11 @@ def __init__(self, deployment):
...
+        from urllib3 import Retry
+        azul_retries = Retry(status_forcelist=[500, 502, 503, 504])
```
Comment (hannes-ucsc) [inline tests/azul_agent.py]: Prefer tuples over lists whenever possible.

File: tests/azul_agent.py (hunk 2 above)
Comment (hannes-ucsc) [reply tests/azul_agent.py]: And I'd call the attribute just `session` or `requests_session` or, actually, `https_session`.

File: tests/azul_agent.py [hunk 4]
```diff
@@ -8,10 +8,15 @@ class AzulAgent:
...
+        self.https_session = requests.Session()
 
     def get_specimen_by_project(self, project_shortname):
         filters = {"file": {"project": {"is": [project_shortname]}}}
         path = '/repository/specimens?filters=' + urllib.parse.quote(json.dumps(filters))
-        response = requests.get(self.azul_url + path)
+        from urllib3 import Retry
+        azul_retries = Retry(status_forcelist=(500, 502, 503, 504))
+        azul_adapter = requests.adapters.HTTPAdapter(max_retries=azul_retries)
+        self.https_session.mount('https://', azul_adapter)
```
Comment (hannes-ucsc) [inline tests/azul_agent.py]: These four lines are once per session, not once per request and should also be in __init__.

**Comment by hannes-ucsc** [comment]
Approved. The build passed. The next push will have https://github.com/HumanCellAtlas/dcp/pull/113/commits/f60874fb3abce80a9eb6ee18de518013e6ed9784 removed and will therefore fail.

**Pull request merged**